src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -955,33 +1013,199 @@ partial class C
         }
 
         [Fact]
-        public void ModifierDifference_01()
+        public void Semantics_RefKind()
         {
-            // access modifier on declaration but not implementation
             var source = """
+                using System;
+
+                var c = new C();
+                ref int i = ref c.P1;
+                c.P1++;
+                Console.Write(i);
+                Console.Write(c.P2);
+                
                 partial class C
                 {
-                    public partial int P { get; set; }
+                    public partial ref int P1 { get; }
+                    public partial ref readonly int P2 { get; }
+
+                    private int _p;
+                    public partial ref int P1 => ref _p;
+                    public partial ref readonly int P2 { get => ref _p; }
                 }
+                """;
+
+            var verifier = CompileAndVerify(source, expectedOutput: "11");
+            verifier.VerifyDiagnostics();
+            var comp = (CSharpCompilation)verifier.Compilation;
 
+            var members = comp.GetMember<NamedTypeSymbol>("C").GetMembers().SelectAsArray(m => m.ToTestDisplayString());
```

Comment:
>.SelectAsArray(m => m.ToTestDisplayString()) [](http://example.com/codeflow?start=75&length=44)

nit: we have a shorthand for this `.ToTestDisplayStrings()` :-)

Reply:
Using this helper changes the type of `members` from `ImmutableArray<string>` to `string[]?` and introduces an ambiguity in AssertEx.Equal. Which I boiled down to the following:

```cs
using System.Collections.Generic;
using System.Collections.Immutable;

class C
{
    public void M(ImmutableArray<string> arr, IEnumerable<string> e) { }
    public void M(ReadOnlySpan<string> span1, ReadOnlySpan<string> span2) { }

    void M0(string[] arr)
    {
        M(["a"], arr); // CS0121: The call is ambiguous between the following methods or properties: 'C.M(ImmutableArray<string>, IEnumerable<string>)' and 'C.M(ReadOnlySpan<string>, ReadOnlySpan<string>)
    }
}
```

I feel like we are trying so hard to avoid actually introducing a rule to make the conversion from array to span better than the conversion from array to interface type..but..that's another conversation.